### ISSUE TITLE
First launch takes much longer after a released update

### ISSUE THREAD

**Issue by ohzee00** [Yellow-Dog-Man/Resonite-Issues#2671]
### Describe the bug?

When launching Resonite after a update releases, it seems to take much longer during the initial loading, getting "stuck" on a particular part of the loading process compared to a second launch.

### To Reproduce

I'm unsure how I can replicate this properly given I don't know what conditions I can do to replicate however I did do a first startup and second startup.

After a update releases:
1. Start up Resonite
2. Notice how it is taking a fair bit of time on a certain part of the loading indicator
3. Shut down Resonite after it is done loading (I judge this by cloud home loading)
4. Start up Resonite again
5. Notice it is much faster

### Expected behavior

Resonite should have semi-consistent load times regardless if there is a update or not.

### Screenshots

![Resonite_2024-07-25_18-19-03](https://github.com/user-attachments/assets/3ac03d1f-39b5-4701-a38b-078b6ee8d043)
During first startup I notice this is the part that it sticks on, not sure if it helps however.

On second startup it just breezes past.

### Resonite Version Number

2024.7.25.1284

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

[First Startup After Update - 2024.7.25.1284 - 2024-07-25 18_18_08 .log](https://github.com/user-attachments/files/16384214/First.Startup.After.Update.-.2024.7.25.1284.-.2024-07-25.18_18_08.log)

[Second startup after update - 2024.7.25.1284 - 2024-07-25 18_21_13.log](https://github.com/user-attachments/files/16384215/Second.startup.after.update.-.2024.7.25.1284.-.2024-07-25.18_21_13.log)

### Additional Context

This could be wrong but I could also swear that the Contacts and Worlds tab stutter a lot worse when on first launch after a update, compared to the second launch. More testing would have to be done for this however, and I'm not sure if its related to this issue.

### Reporters

ohzee

**Comment by Nytra** [comment]
I also noticed a much longer load time after updating

**Comment by ohzee00** [comment]
I did realize something in the logs, I did another launch and I am noticing that the order of the datamodel assembly differs between the first launch and second, with the third launch being the same as the second.

Could the order signify something?

Here is the third launch, notice how on line 52 it loads it in the same order as the above log but _not_ the first.
[DESKTOP-V75BHJO - 2024.7.25.1284 - 2024-07-25 18_32_17.log](https://github.com/user-attachments/files/16384257/DESKTOP-V75BHJO.-.2024.7.25.1284.-.2024-07-25.18_32_17.log)

**Comment by Frooxius** [comment]
@ohzee00 No, the order differing is normal. They're processed in parallel on multiple threads, so there's some randomness, but that won't affect this.

**Comment by ohzee00** [comment]
Hopefully this is more helpful, I am noticing the AssemblyMetadata folder, which I assume has the precomputed hashes, is being modified on first launch. I do not see it happening on the second.

For testing I delete the metadata and allow steam to verify the install to replace them thus replicating a long startup as if I just updated. (Which does happen!)

This is one of the files that steam provides (changed to .txt because apparently I can't upload it):
[Assembly-CSharp.txt](https://github.com/user-attachments/files/16384325/Assembly-CSharp.txt)

This is the same one that is modified on first startup:
[Assembly-CSharp.txt](https://github.com/user-attachments/files/16384326/Assembly-CSharp.txt)

Weirder still, I don't see the hash being different at all, just the modified date inside the file.

As a sanity test I copied the second start up .metadata files and allowed steam to verify them, when replacing them with my second startup files it the load times were the same despite being a first startup!
